### PR TITLE
Add no-async-promise-executor lint rule

### LINKED ISSUE #4806: Add no-async-promise-executor lint rule
`"no-async-promise-executor": "error"` should be added to `.eslintrc.json` and all problems listed fixed. All of code failing this check is in `std/node/_fs`

### PR DESCRIPTION
**Pull request by disizali**
closes #4806

### CODE REVIEW

File: std/node/_fs/_fs_appendFile.ts [hunk 1]
```diff
@@ -39,13 +39,13 @@ export function appendFile(
...
         }
-        const file = await Deno.open(pathOrRid, getOpenOptions(flag));
+        const file = Deno.openSync(pathOrRid, getOpenOptions(flag));
         rid = file.rid;
       }
 
       const buffer: Uint8Array = new TextEncoder().encode(data);
 
-      await Deno.write(rid, buffer);
+      Deno.writeSync(rid, buffer);
```
Comment (ry) [inline std/node/_fs/_fs_appendFile.ts:48]: The function is now sync... The goal is to apply this lint rule without changing the behavior.

Comment (disizali) [reply std/node/_fs/_fs_appendFile.ts]: right , all sync methods removed 👍🏻

File: std/node/_fs/_fs_chmod.ts [hunk 2]
```diff
@@ -13,20 +13,9 @@ export function chmod(
...
-    }
-  })
-    .then(() => {
-      callback();
-    })
-    .catch((err) => {
-      callback(err);
-    });
+  Deno.chmod(path, getResolvedMode(mode))
+    .then(() => callback)
```
Comment (ry) [inline std/node/_fs/_fs_chmod.ts:17]: this seems wrong. Shouldn't it be `.then(callback)` ?

File: std/node/_fs/_fs_copy.ts [hunk 3]
```diff
@@ -7,20 +7,9 @@ export function copyFile(
...
-    }
-  })
-    .then(() => {
-      callback();
-    })
-    .catch((err) => {
-      callback(err);
-    });
+  Deno.copyFile(source, destination)
+    .then(() => callback)
```
Comment (ry) [inline std/node/_fs/_fs_copy.ts:11]: ditto

File: std/node/_fs/_fs_chmod.ts (hunk 2 above)
Comment (disizali) [reply std/node/_fs/_fs_chmod.ts:17]: This functions , does'n provide any callback arguments while promise resolved,

Just one Error type argument on catch

According to [node document](https://nodejs.org/api/fs.html)

File: std/node/_fs/_fs_dir.ts [hunk 4]
```diff
@@ -17,6 +17,7 @@ export default class Dir {
   }
 
   read(callback?: Function): Promise<Dirent | null> {
+    // eslint-disable-next-line no-async-promise-executor
```
Comment (bartlomieju) [inline std/node/_fs/_fs_dir.ts:20]: Remove this ignore and async executor below

File: std/node/_fs/_fs_chmod.ts (hunk 2 above)
Comment (bartlomieju) [reply std/node/_fs/_fs_chmod.ts:17]: So it should be `.then(() => callback())`?

File: std/node/_fs/_fs_dir.ts (hunk 4 above)
Comment (disizali) [reply std/node/_fs/_fs_dir.ts]: done 👍

File: std/node/_fs/_fs_chmod.ts (hunk 2 above)
Comment (disizali) [reply std/node/_fs/_fs_chmod.ts]: right , was my hurry , done 👍

File: std/node/_fs/_fs_dir.ts [hunk 5]
```diff
@@ -17,25 +17,24 @@ export default class Dir {
...
-        if (callback) {
-          callback(err, null);
-        }
-        reject(err);
+    return new Promise((resolve, reject) => {
+      if (!this.asyncIterator) {
+        this.asyncIterator = Deno.readdir(this.path)[Symbol.asyncIterator]();
       }
+      this.asyncIterator
+        ?.next()
```
Comment (bartlomieju) [inline std/node/_fs/_fs_dir.ts:25]: `?` is unnecessary here, it should rather be `!`

Comment (disizali) [reply std/node/_fs/_fs_dir.ts]: thanks

Comment (kitsonk) [reply std/node/_fs/_fs_dir.ts]: actually, it should have an `assert(this.asyncIterator)` on the line before! 😄

Comment (kitsonk) [reply std/node/_fs/_fs_dir.ts]: we should always try to avoid not-null assertions where possible

Comment (disizali) [reply std/node/_fs/_fs_dir.ts]: make sense because of `strict null-checking` 🤔 
thank you

**Pull request merged**